src/surface.cc

```diff
@@ -136,8 +136,19 @@ img_data_t SurfaceNode::read_png_or_dat(std::string filename) const
 {
 	img_data_t data;
 	std::vector<uint8_t> png;
-	
-	lodepng::load_file(png, filename);
+	int ret_val = 0;	
+	try{
+		 ret_val = lodepng::load_file(png, filename);
+	}catch(std::bad_alloc &ba){
+		
+		PRINTB("WARNING: bad_alloc caught for '%s'.", ba.what());
+		return data;	
+	}
+
+	if(ret_val == 78){
+		PRINTB("WARNING: The file '%s'. does not exists or is too large to load", filename);
```

Comment:
Now that you're catching the "too large to load" issue, that text shouldn't be part of the error message any longer.
Also, the error should probably say "couldn't be opened" instead of "does not exist", as there are multiple reasons why a file cannot be opened (e.g. wrong permissions).